src/neo/SmartContract/InteropService.cs

```diff
@@ -291,8 +291,8 @@ private static bool Runtime_GetNotifications(ApplicationEngine engine)
                 notifications = notifications.Where(p => p.ScriptHash == hash);
             }
 
-            if (!engine.CheckArraySize(notifications.Count())) return false;
-            engine.CurrentContext.EvaluationStack.Push(notifications.Select(u => new VM.Types.Array(new StackItem[] { u.ScriptHash.ToArray(), u.State })).ToArray());
+            if (notifications.Count() > engine.MaxStackSize) return false;
```

Comment:
Is not the sum of the current count + notifications.count()?

Reply:
Because I think the `ReferenceCounter` will check it later. But we can optimize it before that.

Reply:
It can be a little difficult to optimize it. Because `notifications` is a nested array. Leave it to `ReferenceCounter`.

Reply:
Yes, i thought that the check of this will come later with the `ReferenceCounter`